File: tests/rc_file_test.rs
use dcss_api::Webtile;

#[test]
fn write_read_rc() {
    let mut webtile =
        Webtile::connect("ws://localhost:8080/socket", 0, "0.29").expect("Failed to connect.");

    // Empty message queue;
    while webtile.get_message().is_some() {}

    webtile
        .login_with_credentials("Username", "Password")
        .expect("Login failed.");

    // Empty message queue;
    while webtile.get_message().is_some() {}

    webtile
        .set_rc_file("seeded-web-trunk", "this is a test")
        .expect("Failed to write");

    let rc_file = webtile
        .get_rc_file("seeded-web-trunk")
        .expect("Failed to read.");

    assert_eq!("this is a test", rc_file);

    // Empty message queue;
    while webtile.get_message().is_some() {}

    webtile
        .set_rc_file("seeded-web-trunk", "show_more = false\nrest_delay = -1")
        .expect("Failed to write");

    let rc_file = webtile
        .get_rc_file("seeded-web-trunk")
        .expect("Failed to read.");

    assert_eq!("show_more = false\nrest_delay = -1", rc_file);

    webtile.disconnect().expect("Failed to disconnect");
}

#[test]
fn blank_rc_file() {
    let mut webtile =
        Webtile::connect("ws://localhost:8080/socket", 0, "0.29").expect("Failed to connect.");

    // Empty message queue;
    while webtile.get_message().is_some() {}

    webtile
        .login_with_credentials("Username", "Password")
        .expect("Login failed.");

    // Empty message queue;
    while webtile.get_message().is_some() {}

    webtile
        .set_rc_file("seeded-web-trunk", "")
        .expect("Failed to write");

    let rc_file = webtile
        .get_rc_file("seeded-web-trunk")
        .expect("Failed to read.");

    assert_eq!("", rc_file);

    // Empty message queue;
    while webtile.get_message().is_some() {}

    webtile
        .set_rc_file("seeded-web-trunk", "show_more = false\nrest_delay = -1")
        .expect("Failed to write");

    let rc_file = webtile
        .get_rc_file("seeded-web-trunk")
        .expect("Failed to read.");

    assert_eq!("show_more = false\nrest_delay = -1", rc_file);

    webtile.disconnect().expect("Failed to disconnect");
}

File: python/tests/rc_file_test.py
import pytest
import dcss_api
from dcss_api import APIErr
import json


def test_write_read_rc():
    # Connect to DCSS Webtile
    webtile = dcss_api.WebtilePy("ws://localhost:8080/socket", 0, "0.29")

    while (message := webtile.get_message()) != None:
        pass

    webtile.login_with_credentials("Username", "Password")

    while (message := webtile.get_message()) != None:
        pass

    webtile.set_rc_file("seeded-web-trunk", "this is a test")

    rc_file = webtile.get_rc_file("seeded-web-trunk")

    assert rc_file == "this is a test"

    while (message := webtile.get_message()) != None:
        pass

    webtile.set_rc_file("seeded-web-trunk", "show_more = false\nrest_delay = -1")

    rc_file = webtile.get_rc_file("seeded-web-trunk")

    assert rc_file == "show_more = false\nrest_delay = -1"

    webtile.disconnect()


def test_blank_rc_file():
    # Connect to DCSS Webtile
    webtile = dcss_api.WebtilePy("ws://localhost:8080/socket", 0, "0.29")

    while (message := webtile.get_message()) != None:
        pass

    webtile.login_with_credentials("Username", "Password")

    while (message := webtile.get_message()) != None:
        pass

    webtile.set_rc_file("seeded-web-trunk", "")

    rc_file = webtile.get_rc_file("seeded-web-trunk")

    assert rc_file == ""

    while (message := webtile.get_message()) != None:
        pass

    webtile.set_rc_file("seeded-web-trunk", "show_more = false\nrest_delay = -1")

    rc_file = webtile.get_rc_file("seeded-web-trunk")

    assert rc_file == "show_more = false\nrest_delay = -1"

    webtile.disconnect()
